### PR TITLE
Composite and compositepi

### PR DESCRIPTION
**Pull request by postvakje**
`composite(n)` returns the n-th composite number. 
`compositepi(n)` returns the number of positive composite numbers less than or equal to n.

### CODE REVIEW

**Comment by jtnydv25** [comment]
A much better method for `composite(nth)` would be to use logarithmic integral, like in `prime(nth)`

**Comment by postvakje** [comment]
Thank you for the suggestion.  I was experimenting with prime and primepi from version 1.0 which use the sieve.  I have now rewritten composite to be similar to the new prime().

File: sympy/ntheory/generate.py [hunk 1]
```diff
@@ -741,3 +741,82 @@ def cycle_length(f, x0, nmax=None, values=False):
...
+        if not isprime(a):
+            n_composites -= 1
+        a -= 1
+    if isprime(a):
+        a -= 1
+    return a
+
+
+def compositepi(n):
+    """ Return the number of positive composite numbers less than or equal to n.
```
Comment (asmeurer) [inline sympy/ntheory/generate.py]: Should note here too that 4 is the first composite

File: sympy/ntheory/generate.py [hunk 2]
```diff
@@ -741,3 +741,82 @@ def cycle_length(f, x0, nmax=None, values=False):
...
+        >>> composite(17737)
+        20000
+
+        See Also
+        ========
+
+        sympy.ntheory.primetest.isprime : Test if n is prime
+        primerange : Generate all primes in a given range
+        primepi : Return the number of primes less than or equal to n
+        prime : Return the nth prime
```
Comment (asmeurer) [inline sympy/ntheory/generate.py]: Mention conpositepi

**Comment by asmeurer** [comment]
Didn't notice this was finished. Please comment when you a ready for a PR to be reviewed again, as GitHub doesn't send notifications on new commits. 

This looks good to me. I'm merging.

**Pull request merged**